inc/Fields/Honeypot.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace AntispamBee\Fields;
+
+use DOMDocument;
+use DOMXPath;
+
+class Honeypot {
+	protected $salt;
+
+	/**
+	 * $options['form_id'];
+	$options['form_name'];
+	$options['field_type'];
+	$options['field_id'];
+	$options['field_name'];
+	 * @param $markup
+	 * @param $options
+	 */
+	public static function inject( $markup, $options ) {
+		$dom = new DOMDocument();
+		$dom->loadHTML( $markup, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+		$xpath = new DOMXPath( $dom );
+		$input = $xpath->query( '//*[@id="' . $options['field_id'] . '"]' )->item( 0 );
+		if ( ! $input ) {
```

Comment:
@angelocali Are we sure `ext-dom` is available by default on hosts? :thinking: If not, but we will need it, we should add a check to `antispam_bee.php` so we make sure the plugin is not loaded if the extension is missing.

Reply:
@2ndkauboy From the [WordPress Handbook](https://make.wordpress.org/hosting/handbook/server-environment/#php-extensions) I see that `ext-dom` is strongly recommended, but WordPress also works without it. Would you rather deactivate the plugin, rule or try an alternative approach without DOMDocument if the extension isn’t available?

Reply:
I've checked with PHP7.4 and `ext-dom` is not included in `php7.4-common`, so I've added another check in 5d189b24cc616a3855e38c3a3490c09566eed1f5